embassy-stm32/src/usart/mod.rs

```diff
@@ -303,6 +311,397 @@ impl<'d, T: BasicInstance, TxDma, RxDma> Uart<'d, T, TxDma, RxDma> {
     }
 }
 
+pub struct UartWithIdle<'d, T: BasicInstance, TxDma = NoDma, RxDma = NoDma> {
+    tx: UartTx<'d, T, TxDma>,
+    rx: UartRxWithIdle<'d, T, RxDma>,
+}
```

Comment:
Should we make this API consistent with the recent nRF changes and avoid having a UartWithIdle struct - just have a split_woth_idle method on Uart. On nRF, this reduced a lot of code duplication also.

Reply:
Thanks for reminding me this recent change on the nRF side. I saw it but forgot about it. I pushed a new commit where I remove UartWithIdle and add split_with_idle to Uart. I also added the possibility to only create UartRxWithIdle directly which can save 1 pin at the expanse of some code duplication but I think it is worth it.